### PR TITLE
Add priority to DSCP spec

### PR DESCRIPTION
**Pull request by alvestrand**
This completes the move of "priority" from the webrtc-pc spec to the spec formerly known as DSCP.
It needs the help of reviewers who can advise on Bikeshed syntax, since a number of constructs are still in the HTML-based format, and linking is a mess.

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/webrtc-dscp-exp/pull/5.html" title="Last updated on Jan 16, 2020, 1:44 PM UTC (3d78384)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/webrtc-dscp-exp/5/c89b908...3d78384.html" title="Last updated on Jan 16, 2020, 1:44 PM UTC (3d78384)">Diff</a>

### CODE REVIEW

File: index.bs [hunk 1]
```diff
@@ -8,46 +8,222 @@ Repository: w3c/webrtc-dscp-exp
...
+              </tr>
+            </tbody>
+          </table>
+        </div>
+      </section>
+      <p>Applications that use this API should be aware that often better
+      overall user experience is obtained by lowering the priority of things
+      that are not as important rather than raising the priority of the things
+      that are.</p>
+    </section>
```
Comment (henbos) [inline index.bs]: The indentation seems a bit aggressive here compared to the surrounding

File: index.bs [hunk 2]
```diff
@@ -8,46 +8,222 @@ Repository: w3c/webrtc-dscp-exp
...
+
+            <dt data-tests="RTCPeerConnection-createDataChannel.html"><code>priority</code> of type <span class=
+            "idlAttrType"><a>RTCPriorityType</a></span>, readonly</dt>
+            <dd>
+              <p>The <code>priority</code> attribute returns the priority for
+              this <code><a>RTCDataChannel</a></code>. The priority is assigned
+              by the user agent at channel creation time. On getting, the
+              attribute MUST return the value of the
+              <a>[[\DataChannelPriority]]</a> slot.</p>
+            </dd>
```
Comment (henbos) [inline index.bs]: Indentation?

File: index.bs [hunk 3]
```diff
@@ -8,46 +8,222 @@ Repository: w3c/webrtc-dscp-exp
...
+</dl>
+## New DataChannelInit members
+<dl>
+            <dt data-tests="RTCPeerConnection-createDataChannel.html"><code>priority</code> of type <span class=
+            "idlMemberType"><a>RTCPriorityType</a></span>, defaulting to
+            <code>low</code></dt>
+            <dd>
+              <p>Priority of this channel.</p>
+            </dd>
```
Comment (henbos) [inline index.bs]: Indentation?

File: index.bs [hunk 4]
```diff
@@ -8,46 +8,222 @@ Repository: w3c/webrtc-dscp-exp
...
+            <tr>
+              <td>257 to 512</td>
+              <td><code><a data-link-for="RTCPriorityType">medium</a></code></td>
+            </tr>
+            <tr>
+              <td>513 and greater</td>
+              <td><code><a data-link-for="RTCPriorityType">high</a></code></td>
+            </tr>
+          </table>
+        </li>
```
Comment (henbos) [inline index.bs]: Indentation?

**Comment by alvestrand** [comment]
Indentation issues are part of nuking the non-bikeshed markup.
I need help (I think) with figuring out what I'm doing wrong wrt bikeshed.

File: index.bs [hunk 5]
```diff
@@ -8,46 +8,222 @@ Repository: w3c/webrtc-dscp-exp
...
+      that are not as important rather than raising the priority of the things
+      that are.</p>
+    </section>
+
+## Extension to EncodingParameters {#encoding-parameters}
 
 <pre class='idl'>
 partial dictionary RTCRtpEncodingParameters {
-    RTCPriorityType networkPriority;  // Note: No default
+  RTCPriorityType priority = "low";
```
Comment (alvestrand) [inline index.bs]: Should remove default here; precedence needs to be explicit, and easier to specify with no default.

**Comment by dontcallmedom** [comment]
@alvestrand I think I've fixed all the main auto-linking issues

File: index.bs [hunk 6]
```diff
@@ -8,46 +8,190 @@ Repository: w3c/webrtc-dscp-exp
...
+  RTCPriorityType priority;
+  RTCPriorityType localPriority;
+  RTCPriorityType networkPriority;
 };
 </pre>
 
-networkPriority definition {#networkPriority}
-----
+<dl>
+  <dt><dfn dict-member for=RTCRtpEncodingParameters><code>priority</code></dfn> of type {{RTCPriorityType]], defaulting to
```
Comment (jan-ivar) [inline index.bs]: }}

File: index.bs [hunk 7]
```diff
@@ -8,46 +8,190 @@ Repository: w3c/webrtc-dscp-exp
...
+</pre>
+## New RTCDataChannel attribute
+<dl>
+  <dt><dfn for=RTCDataChannel attribute>priority</dfn> of type {{RTCPriorityType}}, readonly</dt>
+  <dd>
+    <p>The <code>priority</code> attribute returns the priority for
+      this {{RTCDataChannel}}. The priority is assigned
+      by the user agent at channel creation time. On getting, the
+      attribute MUST return the value of the
+      [[\DataChannelPriority]] slot.</p>
```
Comment (jan-ivar) [inline index.bs]: The`\` here is visible in the document, and formatting doesn't match uses elsewhere.

File: index.bs [hunk 8]
```diff
@@ -8,46 +8,190 @@ Repository: w3c/webrtc-dscp-exp
...
+overall user experience is obtained by lowering the priority of things
+that are not as important rather than raising the priority of the things
+that are.
 
+## Extension to RTCRtpEncodingParameters ## {#encoding-parameters}
 <pre class='idl'>
 partial dictionary RTCRtpEncodingParameters {
-    RTCPriorityType networkPriority;  // Note: No default
+  RTCPriorityType priority;
+  RTCPriorityType localPriority;
```
Comment (jan-ivar) [inline index.bs]: This is more than a move, adding a new `localPriority` field that did not exists in webrtc-pc.

Having three knobs to control two things seems like one too many (one of the three seem redundant).

E.g. I could already accomplish all combinations using `priority` and `networkPriority` alone.

Comment (alvestrand) [reply index.bs]: Yes. For symmetry, I would prefer to have localPriority and networkPriority only, but the 3-way thing preserves the text that had consensus with the IETF a long time ago (one control that influences both things).

The way to accomplish setting localPriority only without the value (setting priority to the desired value and also setting networkPrioirty to the default value) seems unintuitive and convoluted.

Suggestions?

Comment (jan-ivar) [reply index.bs]: I'd pick 1 or 2 I think. I don't understand when I would want to set different priorities.

**Comment by alvestrand** [comment]
Removed localPriority, submitting based on approvals.

**Pull request merged**